thingy53: samples and applications: Clean USB CDC related configs

mkapala-nordic opened the pull request:
Thingy:53 board in the NCS already enables logging over USB CDC by default.
There is no need anymore to explicitly enable most of USB CDC related options in Thingy:53 app configs.

Jira: NCSDK-18596

Signed-off-by: Mateusz Kapala <mateusz.kapala@nordicsemi.no>

NordicBuilder commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| zephyr | https://github.com/nrfconnect/sdk-zephyr/commit/e2bec540218d91ac8ce5015cc101b3bdc701eaef | https://github.com/nrfconnect/sdk-zephyr/pull/1043 | [nrfconnect/sdk-zephyr#1043/files](https://github.com/nrfconnect/sdk-zephyr/pull/1043/files) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10115/1/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10115/1/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| desktop52_verification | X | | |
| test-fw-nrfconnect-apps | X | | |
| test-fw-nrfconnect-ble | X | | |
| test-fw-nrfconnect-ble_mesh | X | | |
| test-fw-nrfconnect-ble_samples | X | | |
| test-fw-nrfconnect-chip | X | | |
| test-fw-nrfconnect-nfc | X | | |
| test-fw-nrfconnect-nrf-iot_lwm2m | X | | |
| test-fw-nrfconnect-nrf-iot_thingy91 | X | | |
| test-fw-nrfconnect-nrf_crypto | X | | |
| test-fw-nrfconnect-rpc | X | | |
| test-fw-nrfconnect-rs | X | | |
| test-fw-nrfconnect-tfm | X | | |
| test-fw-nrfconnect-thread | X | | |
| test-fw-nrfconnect-zigbee | X | | |
| test-sdk-find-my | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10115/1/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-10115
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

kkasperczyk-no reviewed (commented):
Could you apply the same changes as in matter_weather_station/.../prj.conf to the prj_factory_data.conf?

MarekPieta commented on applications/machine_learning/src/modules/Kconfig.usb_state:
```diff
@@ -12,6 +12,14 @@ menuconfig ML_APP_USB_STATE
 	help
 	  Enable USB subsystem on application start.
 
+if ML_APP_USB_STATE
```
I think we should consider disabling the `CONFIG_BOARD_SERIAL_BACKEND_CDC_ACM` in Thingy:53 MCUboot configuration. Actually it should not be needed for the MCUboot

MarekPieta replied:
We also should definitely disable one of the USB CDC ACM instances in Thingy:53 MCUboot's DTS (after your changes you have 2 USB CDC instanced and one of them is unused)

MarekPieta replied:
On instance is added by Thingy:53 board and the second one comes from `usb.overlay` applied automatically in the `nrf/modules/mcuboot/CMakelists.txt` file

mkapala-nordic replied:
Discussed offline, taken steps:
+ sdk-mcuboot noup change where `CONFIG_BOARD_SERIAL_BACKEND_CDC_ACM` is disabled in thingy board config https://github.com/nrfconnect/sdk-mcuboot/pull/237
+ sdk-zephyr noup change where `CONFIG_MCUBOOT_USB_SUPPORT` is disabled in thingy board config https://github.com/nrfconnect/sdk-zephyr/pull/1043

MarekPieta commented on samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf:
```diff
@@ -59,22 +59,15 @@ CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
 
-CONFIG_LOG_BACKEND_UART=y
 CONFIG_LOG_PRINTK=y
 CONFIG_LOG_PROCESS_THREAD_STACK_SIZE=1024
```
Do we need to configure these explicitly? I think we should ensure that the USB CDC configuration would work out-of-the-box also for other samples (that rely on default values here).

MarekPieta replied:
I would opt for removing all of the unnecessary configurations related to logging or USB CDC ACM to simplify the user experience (if something is mandatory we should probably set it by default in Thingy:53 board definition files).

mkapala-nordic replied:
Removed all unnecessary configs

nordicjm commented:
Needs to be rebased to include west.yml changes and changes in https://github.com/nrfconnect/sdk-nrf/pull/9999 which removes some files touched here

mkapala-nordic commented:
> Could you apply the same changes as in matter_weather_station/.../prj.conf to the prj_factory_data.conf?

@kkasperczyk-no Of course, i must have missed them at the first time, fixed.

> Needs to be rebased to include west.yml changes and changes in #9999 which removes some files touched here

Rebased onto #9999.

mkapala-nordic commented on applications/nrf5340_audio/src/modules/dfu_entry.c:
```diff
@@ -18,19 +18,6 @@
 #include "macros_common.h"
 #include "channel_assignment.h"
 
-#ifdef CONFIG_MCUMGR_CMD_OS_MGMT
```
I talked with @kapi-no, @MarekPieta how we should proceed.
We've come to conclusion that this PR can be independently merged before #9999. 
I'll rebase back to main and drop all changes from mentioned PR.

MarekPieta commented on samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf:
```diff
@@ -59,22 +59,13 @@ CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
-CONFIG_LOG_PROCESS_THREAD_STACK_SIZE=1024
-CONFIG_LOG_PROCESS_THREAD_STARTUP_DELAY_MS=4000
 CONFIG_USE_SEGGER_RTT=n
 CONFIG_LOG_BACKEND_RTT=n
```
IIRC we could actually provide logs both over CDC ACM and RTT (you can double check if it has no side effects).

mkapala-nordic replied:
Both backends can work simultaneously so i just left enabling UART backend here.

MarekPieta commented on samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp.conf:
```diff
@@ -59,22 +59,13 @@ CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
-CONFIG_USB_CDC_ACM_LOG_LEVEL_ERR=y
-CONFIG_USB_DEVICE_LOG_LEVEL_ERR=y
 
 CONFIG_LOG_BACKEND_UART=y
```
Please comment on why this is still needed

mkapala-nordic commented on samples/bluetooth/peripheral_uart/boards/thingy53_nrf5340_cpuapp_ns.conf:
```diff
@@ -59,22 +59,13 @@ CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
-CONFIG_USB_CDC_ACM_LOG_LEVEL_ERR=y
-CONFIG_USB_DEVICE_LOG_LEVEL_ERR=y
 
 CONFIG_LOG_BACKEND_UART=y
```
Added comment

mkapala-nordic commented on applications/machine_learning/src/modules/Kconfig.usb_state:
```diff
@@ -12,6 +12,14 @@ menuconfig ML_APP_USB_STATE
 
+if ML_APP_USB_STATE
+
+# Let USB device initialization to the USB module
```
Changed comment to Kconfig help

mkapala-nordic commented:
Rebase to newest main

mkapala-nordic commented:
Rebased + CI rerun to fix compliance check fail from sdk-zephyr PR

nordicjm commented on applications/zigbee_weather_station/configuration/thingy53_nrf5340_cpuapp/prj.conf:
```diff
@@ -11,15 +11,12 @@
+CONFIG_USB_DEVICE_INITIALIZE_AT_BOOT=n
 CONFIG_USB_DEVICE_PRODUCT="Thingy 53: Zigbee Weather Station"
 CONFIG_USB_CDC_ACM_LOG_LEVEL_WRN=y
-CONFIG_USB_DEVICE_VID=0x1915
```
not addressed

mkapala-nordic replied:
@adsz-nordic, please take a look at @MarekPieta comment

sebastiandraus replied:
Should be fine to use common USB PID, previously default Zephyr's one was used.

mkapala-nordic commented:
Updated manifest after merging sdk-zephyr and sdk-mcuboot related PRs

mkapala-nordic commented:
Rebased due to manifest conflict
